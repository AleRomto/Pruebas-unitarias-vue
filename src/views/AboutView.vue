<template>
    <div>
      <h1>About Us</h1>
      <p>Soy un about</p>
    </div>
  </template>