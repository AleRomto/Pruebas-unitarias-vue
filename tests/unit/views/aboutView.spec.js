import { mount } from '@vue/test-utils'
import { createRouter, createWebHistory } from 'vue-router'
import AboutView from '@/views/AboutView.vue'

describe('AboutView', () => {
  test('debe renderizar el componente AboutView correctamente', async () => {
    const router = createRouter({
      history: createWebHistory(),
      routes: [{
        path: '/about',
        name: 'about',
        component: AboutView
      }]
    })
    router.push('/about')
    await router.isReady()

    const wrapper = mount(AboutView, {
      global: {
        plugins: [router]
      }
    })

    expect(wrapper.findComponent(AboutView).exists()).toBe(true)
  })

  // agregar el snapshot 
  test('debe coincidir con el snapshot', async () => {
    const router = createRouter({
      history: createWebHistory(),
      routes: [{
        path: '/about',
        name: 'about',
        component: AboutView
      }]
    })
    router.push('/about')
    await router.isReady()

    const wrapper = mount(AboutView, {
      global: {
        plugins: [router]
      }
    })

    // Prueba del snapshot
    expect(wrapper.html()).toMatchSnapshot()
  })
})
