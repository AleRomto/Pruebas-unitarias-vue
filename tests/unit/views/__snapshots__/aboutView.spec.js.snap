// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutView debe coincidir con el snapshot 1`] = `
<div>
  <h1>About Us</h1>
  <p>Soy un about</p>
</div>
`;
